### PR TITLE
Notification UI

### PR DESCRIPTION
**Pull request by bliitzkrieg**
## Overview
- Adds UI button to Incident report page that opens a modal to notify users.
- Adds Proxy endpoint to authenticate NFORS and forward payload to authenticated NFORS endpoint

## GitHub Issues
- #438

## Changes
- Updated Readme with additional requirement details
- Added UI button to Incident report page
- Added proxy endpoint to proxy NFORS with authentication

## Screenshots / Videos
Screen Shot 2020-04-02 at 6 35 52 PM

Screen Shot 2020-04-07 at 3 30 15 PM

Screen Shot 2020-04-07 at 3 30 19 PM

## Steps to Test
- To be determined

### CODE REVIEW

File: client/app/notify/notification.controller.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      .map(apparatus => apparatus.personnel)
+      .flat()
+      .filter(personnel => personnel)
+      .map(personnel => personnel.employee_id);
+
+    this.payload = {
+      firecaresId: currentPrincipal.FireDepartment.firecares_id,
+      incident_number: incidentData.incident.description.incident_number,
+      requestor_name: currentPrincipal.name,
+      test: true,
```
Comment (bliitzkrieg) [inline client/app/notify/notification.controller.js:35]: I hardcoded test = true so we can test in production without sending notifications to ensure this works. The notifications get logged in Notification table on NFORS so we can verify its working as expected.

File: server/api/notify/notify.controller.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import axios from 'axios';
+
+export async function notify(req, res) {
+  try {
+    const { data } = await axios.post(process.env.NFORS_API_URL + 'auth/local', {
+      email: process.env.NFORS_AUTHENTICATION_USERNAME,
+      password: process.env.NFORS_AUTHENTICATION_PASSWORD
```
Comment (bliitzkrieg) [inline server/api/notify/notify.controller.js:7]: These variables need to be added to the `env.json` in production as well

File: client/app/notify/notification.controller.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      .map(apparatus => apparatus.personnel)
+      .flat()
+      .filter(personnel => personnel)
+      .map(personnel => personnel.employee_id);
+
+    this.payload = {
+      firecaresId: currentPrincipal.FireDepartment.firecares_id,
+      incident_number: incidentData.incident.description.incident_number,
+      requestor_name: currentPrincipal.name,
+      test: process.env.NOTIFICATION_TEST_MODE,
```
Comment (bliitzkrieg) [inline client/app/notify/notification.controller.js:35]: I set this to an environment variable so we can test in production without sending notifications. The notifications get logged in Notification table on NFORS so we can verify its working as expected.

**Comment by chopchop505** [comment]
Can you remove the "has been added to your career diary"?
Maybe something like, "This incident as been identified as a potential exposure by your department.  Please be sure to log any exposures"

File: server/api/notify/index.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import * as auth from '../../auth/auth.service';
+import * as controller from './notify.controller';
+
+const router = new Router();
+
+router.post(
+  '/',
+  bodyParser.json(),
+  auth.isApiAuthenticated,
+  auth.hasRole('user'),
```
Comment (chopchop505) [inline server/api/notify/index.js:15]: change to   auth.hasRole('department_admin'),

**Comment by chopchop505** [comment]
If the user does not select units or personnel, the notification should go out to all users that have assigned themselves to the incident.  

The units and personnel filter should only be used to limit the audience.
Thoughts on making this clearer? @bingles 

image

**Comment by chopchop505** [comment]
Maybe even just a radio button

Option A: Notify all users
Option B: Notify a subset of personnel/units 

When Option B is selected, then you would show the Units and Personnel drop down.

**Comment by bliitzkrieg** [comment]
The changes requested are implemented. The default message was what we used in NFORS. Should I change the copy there as well?

**Pull request merged**